File: app/src/main/java/ru/magflayer/spectrum/presentation/pages/main/history/HistoryView.java
package ru.magflayer.spectrum.presentation.pages.main.history;

import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import java.util.List;

import ru.magflayer.spectrum.domain.entity.ColorPhotoEntity;
import ru.magflayer.spectrum.presentation.common.mvp.view.PageView;

interface HistoryView extends PageView {

    @StateStrategyType(AddToEndSingleStrategy.class)
    void showHistory(List<ColorPhotoEntity> history);

    @StateStrategyType(OneExecutionStateStrategy.class)
    void openPickPhoto();

}

File: app/src/main/java/ru/magflayer/spectrum/presentation/pages/main/history/HistoryView.kt
package ru.magflayer.spectrum.presentation.pages.main.history

import moxy.viewstate.strategy.AddToEndSingleStrategy
import moxy.viewstate.strategy.OneExecutionStateStrategy
import moxy.viewstate.strategy.StateStrategyType
import ru.magflayer.spectrum.domain.entity.ColorPhotoEntity
import ru.magflayer.spectrum.presentation.common.mvp.view.PageView

interface HistoryView : PageView {

    @StateStrategyType(AddToEndSingleStrategy::class)
    fun showHistory(history: List<ColorPhotoEntity>)

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun openPickPhoto()

}
